update client class loading server configuration

ryankwilliams opened the pull request:
## What does this implement/fix? Explain your changes

This PR updates the client class to have the same capabilities of
loading server configuration settings just like you can from the
command line. Previously it would only load server configuration
settings when instantiating the client class.

Updated example scripts to no longer need to define the default config
settings. These will be set by default.

## Does this close any currently open issues?

No open issues are related to this PR.

ryankwilliams committed:
update client class loading server configuration

This commit updates the client class to have the same capabilities of
loading server configuration settings just like you can from the
command line. Previously it would only load server configuration
settings when instantiating the client class.

Updated example scripts to no longer need to define the default config
settings. These will be set by default.

tvieira commented on miqcli/api.py:
```diff
@@ -276,8 +276,16 @@ def __init__(self, conf, verbose=False):
         """
 
+        # lets first load the correct server configuration settings
+        config = Config(defaults=DEFAULT_CONFIG, verbose=verbose)
```
`defaults=` does not match with what it is. This is actual `settings` or something else, which, by default, receives DEFAULT_CONFIG. but I would call it something other than `defaults`. Does it make sense?

ryankwilliams replied:
Yes that does make sense. I would prefer `settings` over `defaults`. That parameter name has been set and I was just re-using it. Let me make an update to modify this parameter name to better map to what it actual is accepting.

ryankwilliams committed:
rename config class parameter for settings input

Previously the config class accepted settings as a parameter stored into
the variable named defaults. Defaults name does not match with what the
actual parameter data is. Modified the config class to switch the
parameter name to align with the data it can accept.

vi-patel reviewed (approved):
looks good.

tvieira reviewed (approved):
LGTM
